pyatlan/pkg/models.py

```diff
@@ -0,0 +1,300 @@
+    certified: bool = True
+    preview: bool = False
+    connector_type: Optional[AtlanConnectorType] = None
+    category: str = "custom"
+    outputs: dict[str, str] = Field(default_factory=dict)
+    _pkg: PackageDefinition = PrivateAttr()
+    _name: str = PrivateAttr()
+
+    def _init_private_attributes(self) -> None:
+        super()._init_private_attributes()
+        self._pkg = PackageDefinition(
+            package_id=self.package_id,
+            package_name=self.package_name,
+            description=self.description,
+            icon_url=self.icon_url,
+            docs_url=self.docs_url,
+            keywords=self.keywords,
+            allow_schedule=self.allow_schedule,
+            certified=self.certified,
+            preview=self.preview,
+            connector_type=self.connector_type,
+            category=self.category,
+        )
+        self._name = self.package_id.replace("@", "").replace("/", "-")
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def packageJSON(self) -> str:
+        return self._pkg.packageJSON
+
+    @staticmethod
+    def indexJS() -> str:
+        return textwrap.dedent(
+            """\
+        function dummy() {
+            console.log("don't call this.")
+        }
+        module.exports = dummy;
+        """
+        )
+
+    @staticmethod
+    def create_package(pkg: "CustomPackage", args: list[str]):
+        path = args[0]
+        root_dir = Path(path) / pkg.name
+        root_dir.mkdir(parents=True, exist_ok=True)
+        with (root_dir / "index.js").open("w") as index:
+            index.write(CustomPackage.indexJS())
+        with (root_dir / "package.json").open("w") as package:
+            package.write(pkg.packageJSON)
+        config_maps_dir = root_dir / "configmaps"
+        config_maps_dir.mkdir(parents=True, exist_ok=True)
+        templates_dir = root_dir / "templates"
+        templates_dir.mkdir(parents=True, exist_ok=True)
+
+
+class PackageWriter(BaseModel):
+    path: Path
+    pkg: CustomPackage
+    _root_dir: Path = PrivateAttr()
+    _config_maps_dir: Path = PrivateAttr()
+    _templates_dir: Path = PrivateAttr()
+    _env: Environment = PrivateAttr()
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        self._root_dir = self.path / self.pkg.name
+        self._config_maps_dir = self._root_dir / "configmaps"
+        self._templates_dir = self._root_dir / "templates"
+        self._env = Environment(  # noqa: S701
+            loader=PackageLoader("pyatlan.pkg", "templates")
+        )
+        self._env.globals.update({"isinstanceof": isinstance})
+
+    def create_package(self):
+        self._root_dir.mkdir(parents=True, exist_ok=True)
+        with (self._root_dir / "index.js").open("w") as index:
+            index.write(CustomPackage.indexJS())
+        with (self._root_dir / "index.js").open("w") as index:
+            index.write(CustomPackage.indexJS())
```

Comment:
Looks like we're doing this twice, back-to-back — copy/paste mistake (?)

Reply:
fixed